meteor/server/api/rundownLayouts.ts

```diff
@@ -88,6 +80,39 @@ PickerPOST.route('/shelfLayouts/upload/:showStyleBaseId', async (params, req: In
 	res.end(content)
 })
 
+/**
+ * Uploads the layout to the first Show Style Base that uses the given blueprint id
+ */
+PickerPOST.route(
+	'/shelfLayouts/uploadByBlueprintId/:showStyleBlueprintId',
+	async (params, req: IncomingMessage, res: ServerResponse) => {
+		res.setHeader('Content-Type', 'text/plain')
+
+		const showStyleBlueprintId: BlueprintId = protectString(params.showStyleBlueprintId)
+
+		check(showStyleBlueprintId, String)
+
+		const showStyleBases = await fetchShowStyleBasesLight({ blueprintId: showStyleBlueprintId })
+
+		let content = ''
+		try {
+			if (!showStyleBases.length)
+				throw new Meteor.Error(404, `Show Style Base not found for blueprint "${showStyleBlueprintId}"`)
+
+			const body = validateAndGetRestoredLayoutBody(req)
+			await upsertLayout(body, showStyleBases[0])
+
+			res.statusCode = 200
+		} catch (e) {
+			res.statusCode = 500
+			content = e + ''
```

Comment:
The status code 404 is set in the Meteor.Error that is thrown inside the try-clause. To improve the response status code, could this be extracted and used?

Reply:
I'm not sure I understand correctly. Would you like me to assign the status code from the caught Meteor Error to `res.statusCode`? I'm not sure this would work. Yes, we're setting it to 404, but I don't know what other codes can be encountered if something else throws. According to the docs, it can be a number for legacy reasons and _some built-in Meteor functions such as `check` throw errors with a number in this field_. But I don't know if they all are valid/useful HTTP status codes